### PR TITLE
Export remote_sources metadata

### PR DESCRIPTION
**Pull request by mkosiarc**
Update the metadata format and prepare it for
using multiple remote sources
CLOUDBLD-4453

Signed-off-by: mkosiarc <mkosiarc@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

**Review by ssalatsk — APPROVED** [review]

File: atomic_reactor/plugins/exit_koji_import.py [hunk 1]
```diff
@@ -232,15 +232,32 @@ def set_pnc_build_metadata(self, extra):
     def set_remote_sources_metadata(self, extra):
         remote_source_result = self.workflow.prebuild_results.get(PLUGIN_RESOLVE_REMOTE_SOURCE)
         if remote_source_result:
-            url = remote_source_result['annotations']['remote_source_url']
+            remote_sources_image_metadata = [
+                {"name": remote_source["name"], "url": remote_source["url"]}
+                for remote_source in remote_source_result
+            ]
+            extra["image"]["remote_sources"] = remote_sources_image_metadata
```
Comment (rcerven) [inline atomic_reactor/plugins/exit_koji_import.py:239]: we should set this only if self.workflow.source.config.remote_sources

File: atomic_reactor/plugins/exit_koji_import.py [hunk 2]
```diff
@@ -232,15 +232,32 @@ def set_pnc_build_metadata(self, extra):
...
+                        remote_source["remote_source_tarball"]["filename"],
+                    ],
+                }
+                for remote_source in remote_source_result
+            ]
             remote_source_typeinfo = {
-                KOJI_BTYPE_REMOTE_SOURCES: {
-                    'remote_source_url': url,
-                },
+                KOJI_BTYPE_REMOTE_SOURCES: remote_sources_typeinfo_metadata,
```
Comment (rcerven) [inline atomic_reactor/plugins/exit_koji_import.py:253]: we should set this only if self.workflow.source.config.remote_sources

and keep that 'remote_source_url' when self.workflow.source.config.remote_source

File: atomic_reactor/plugins/pre_resolve_remote_source.py [hunk 3]
```diff
@@ -70,7 +75,10 @@ def run(self):
             self.log.info('Aborting plugin execution: missing Cachito configuration')
             return
 
-        if self.workflow.source.config.remote_sources:
+        if (
```
Comment (rcerven) [inline atomic_reactor/plugins/pre_resolve_remote_source.py:78]: can you change formatting here? this looks odd

if (...........................  looks nicer

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 4]
```diff
@@ -235,8 +236,11 @@ def _get_remote_urls_helper(self, koji_build):
                 dest_name = remote_source['dest']
                 remote_sources_urls.append(remote_source)
 
-            elif archive['type_name'] == 'json' and archive['filename'] == 'remote-source.json':
-                cachito_json_url = os.path.join(remote_sources_path, archive['filename'])
+            elif (
```
Comment (rcerven) [inline atomic_reactor/plugins/pre_fetch_sources.py:239]: can you change formatting here? this looks odd

elif (........................... looks nicer

**Review by rcerven — APPROVED** [review]

File: atomic_reactor/plugins/pre_reactor_config.py [hunk 5]
```diff
@@ -160,6 +160,10 @@ def get_cachito_session(workflow):
     return CachitoAPI(config['api_url'], **api_kwargs)
 
 
+def get_allow_multiple_remote_sources(workflow):
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_reactor_config.py:163]: Should we have `fallback` here like other methods have?

File: atomic_reactor/plugins/pre_reactor_config.py [hunk 6]
```diff
@@ -160,6 +160,10 @@ def get_cachito_session(workflow):
...
+    return get_value(workflow, 'allow_multiple_remote_sources', False)
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_reactor_config.py:164]: Should we have it enabled by default?

**Review by lkolacek — APPROVED** [review]

File: atomic_reactor/plugins/pre_reactor_config.py (hunk 5 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_reactor_config.py:163]: we don't need explicit fallback, that was used before, but we don't need it for new methods

File: atomic_reactor/plugins/pre_reactor_config.py (hunk 6 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_reactor_config.py:164]: better not yet, maybe in the future

File: atomic_reactor/plugins/pre_reactor_config.py (hunk 5 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/pre_reactor_config.py:163]: ok

File: atomic_reactor/plugins/pre_reactor_config.py (hunk 6 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/pre_reactor_config.py:164]: in future it will be the only way, so the whole option will be deprecated

**Comment by MartinBasti** [comment]
Should be the new option documented in administration guide?

**Comment by rcerven** [comment]
> Should be the new option documented in administration guide?

yeah, and we should update also koji.md with those new metadata and explain both formats

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 7]
```diff
@@ -235,8 +236,9 @@ def _get_remote_urls_helper(self, koji_build):
                 dest_name = remote_source['dest']
                 remote_sources_urls.append(remote_source)
 
-            elif archive['type_name'] == 'json' and archive['filename'] == 'remote-source.json':
-                cachito_json_url = os.path.join(remote_sources_path, archive['filename'])
+            elif (archive["type_name"] == "json"
```
Comment (ben-alkov) [inline atomic_reactor/plugins/pre_fetch_sources.py:239]: Is there a reason we're changing to double-quotes here?

**Review by ben-alkov — APPROVED** [review]
LGTM with comment

**Comment by mkosiarc** [comment]
As agreed documentation will be updated during work on CLOUDBLD-4533, merging for now.

**Comment by mkosiarc** [comment]
rn updated: New config option - allow_multiple_remote_sources and updated metadata format for remote sources

**Pull request merged**